src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    private BulldozerProducer bulldozerProducer;
+    private ExcavatorProducer excavatorProducer;
+    private TruckProducer truckProducer;
+
+    public MachineServiceImpl() {
+        bulldozerProducer = new BulldozerProducer();
+        excavatorProducer = new ExcavatorProducer();
+        truckProducer = new TruckProducer();
+    }
+
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        List<? extends Machine> machines = List.of();
```

Comment:
better to new ArrayList instead of List.of()

Reply:
I use this to return an empty List if the class fails all checks

Reply:
It doesn't matter.   `new ArrayList` will create a modified list, that better then List.of()